manager/controllers/default/resource/update.class.php

```diff
@@ -119,7 +119,7 @@ public function process(array $scriptProperties = array()) {
         $this->resourceArray['cacheable'] = intval($this->resourceArray['cacheable']) == 1 ? true : false;
         $this->resourceArray['deleted'] = intval($this->resourceArray['deleted']) == 1 ? true : false;
         $this->resourceArray['uri_override'] = intval($this->resourceArray['uri_override']) == 1 ? true : false;
-        $this->resourceArray['syncsite'] = isset($this->resourceArray['syncsite']) && intval($this->resourceArray['syncsite']) == 1 ? true : $this->context->getOption('syncsite_default', 1, $this->modx->_userConfig);
+        $this->resourceArray['syncsite'] = intval($this->context->getOption('syncsite_default', 1, $this->modx->_userConfig)) == 1 ? true : false;
```

Comment:
I certainly can't argue with that. The thing is, the other settings are saved in the database and get put into the resourceArray. What this block of code is doing is to change them from 1 and 0 to true and false. The syncsite setting is not already in the resourceArray since it's not saved in the database, but it still has to be converted from the integer values in the system setting.


On Jan 26, 2015, at 10:55 AM, Romain Tripault <notifications@github.com> wrote:

> In manager/controllers/default/resource/update.class.php:
> 
> > @@ -119,7 +119,7 @@ public function process(array $scriptProperties = array()) {
> >          $this->resourceArray['cacheable'] = intval($this->resourceArray['cacheable']) == 1 ? true : false;
> >          $this->resourceArray['deleted'] = intval($this->resourceArray['deleted']) == 1 ? true : false;
> >          $this->resourceArray['uri_override'] = intval($this->resourceArray['uri_override']) == 1 ? true : false;
> > -        $this->resourceArray['syncsite'] = isset($this->resourceArray['syncsite']) && intval($this->resourceArray['syncsite']) == 1 ? true : $this->context->getOption('syncsite_default', 1, $this->modx->_userConfig);
> > +        $this->resourceArray['syncsite'] = intval($this->context->getOption('syncsite_default', 1, $this->modx->_userConfig)) == 1 ? true : false;
> Once again, this is just my understanding of the code :
> 
> we grab the modResource object, and make it available as an array
> then, we check for form customization rules, which allow to "override" some fields values (syncsite being one fo those)
> then, we process the array data so it become usable in ExtJS
> Again, my interpretation is that overridden values in FC should have an higher priority (be used/applied) than settings.
> 
> —
> Reply to this email directly or view it on GitHub.
>